test/client.js

```diff
@@ -0,0 +1,158 @@
+'use strict';
+
+const metatests = require('metatests');
+const { Metacom } = require('metacom/lib/client');
+
+let token = '';
+
+const runTests = (client) => {
+  metatests.testAsync('system/introspect', async (test) => {
+    const introspect = await client.socketCall('system')('introspect')([
+      'auth',
+      'console',
+      'example',
+      'files',
+      'test',
+    ]);
+    test.strictSame(introspect?.auth?.restore?.[0], 'token');
+    test.end();
+  });
+
+  metatests.testAsync('example/add', async (test) => {
+    const add = await client.socketCall('example')('add')({ a: 1, b: 2 });
+    test.strictSame(add, 3);
+    test.end();
+  });
+
+  metatests.testAsync('example/citiesByCountry', async (test) => {
+    const cities = await client.socketCall('example')('citiesByCountry')({
+      countryId: 1,
+    });
+    test.strictEqual(cities?.result, 'success');
+    test.strictEqual(Array.isArray(cities?.data), true);
+    test.end();
+  });
+
+  metatests.testAsync('example/customError', async (test) => {
+    try {
+      await client.socketCall('example')('customError')();
+    } catch (customError) {
+      test.errorCompare(customError, new Error('Return custom error', 12345));
+      test.strictEqual(customError?.code, 12345);
+    } finally {
+      test.end();
+    }
+  });
+
+  metatests.testAsync('example/customException', async (test) => {
+    try {
+      await client.socketCall('example')('customException')();
+    } catch (customError) {
+      test.errorCompare(customError, new Error('Custom ecxeption', 12345));
+      test.strictEqual(customError?.code, 12345);
+    } finally {
+      test.end();
+    }
+  });
+
+  metatests.testAsync('example/error', async (test) => {
+    try {
+      await client.socketCall('example')('error')();
+    } catch (Error) {
+      test.errorCompare(Error, new Error('Return error'));
+    } finally {
+      test.end();
+    }
+  });
+
+  metatests.testAsync('example/exception', async (test) => {
+    try {
+      await client.socketCall('example')('exception')();
+    } catch (Error) {
+      test.errorCompare(Error, new Error('Example exception'));
+    } finally {
+      test.end();
+    }
+  });
+
+  metatests.testAsync('example/getClientInfo', async (test) => {
+    try {
+      const info = await client.socketCall('example')('getClientInfo')();
+      test.strictEqual(info?.result?.ip, '127.0.0.1');
+      test.strictEqual(info?.result?.token, token);
+      test.strictEqual(info?.result?.accountId, '2');
+    } catch (Error) {
+      console.log(Error);
```

Comment:
AFAIK metatests themselves will handle unexpected error throw, no need to do it by hand.